### PR TITLE
Search for doi and isbn also in web search

### LINKED ISSUE #7575: Entering an Id at the Web Search should search for the id - and not throw an Exception
JabRef version 5.3 on Mac OS X 10.15.7 Catalina

-  I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. select DBLP for search
2. paste DOI 10.1007/s11192-020-03391-y into the web search field
3. click search

```
org.jabref.logic.importer.FetcherException: An error occurred when parsing the query
	at org.jabref@5.3.218/org.jabref.logic.importer.SearchBasedFetcher.performSearch(Unknown Source)
	at org.jabref@5.3.218/org.jabref.gui.importer.fetcher.WebSearchPaneViewModel.lambda$search$1(Unknown Source)
	at org.jabref@5.3.218/org.jabref.gui.util.BackgroundTask$1.call(Unknown Source)
	at org.jabref@5.3.218/org.jabref.gui.util.DefaultTaskExecutor$1.call(Unknown Source)
	at org.jabref.merged.module@5.3.218/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

### PR DESCRIPTION
**Pull request by Ali96kz**
Fixes #7575 
We won't throw an exception if the user put into web search an ISBN or DOI, instead, we will try to fetch it

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Would be nice if you could adjust your PR title to  make our life as reviewer easier ( we don't have issue numbers in our head)

File: src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java [hunk 1]
```diff
@@ -90,8 +103,34 @@ public void search() {
...
+        task = BackgroundTask.wrap(() -> {
+            Optional<DOI> doi = DOI.parse(getQuery());
+            if (doi.isPresent()) {
+                try {
+                    Optional<BibEntry> bibEntry = doiFetcher.performSearchById(doi.get().getDOI());
+                    if (bibEntry.isPresent()) {
+                        return new ParserResult(Collections.singletonList(bibEntry.get()));
+                    }
+                } catch (FetcherException ignore) {
+                    // Ignored, because it can be something else
```
Comment (koppor) [inline src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java:115]: We should at least log the exception.

File: src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java [hunk 2]
```diff
@@ -90,8 +103,34 @@ public void search() {
...
+
+            Optional<ISBN> isbn = ISBN.parse(getQuery());
+            if (isbn.isPresent()) {
+                try {
+                    Optional<BibEntry> bibEntry = isbnFetcher.performSearchById(isbn.get().getNormalized());
+                    if (bibEntry.isPresent()) {
+                        return new ParserResult(Collections.singletonList(bibEntry.get()));
+                    }
+                } catch (FetcherException ignore) {
+                    // Ignored, because it can be something else
```
Comment (koppor) [inline src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java:127]: We should at least log the exception

**Review by koppor — CHANGES_REQUESTED** [review]
Thank you for the fix. Two small comments, but one large one:

We aim to have JabRef's architecture clean. Thus, there should not be any logic in a view. Proposal: Add a `CompositeFetcher` similar to `org.jabref.logic.importer.fetcher.CompositeSearchFetcher`, which tries the provided fetchers one after another.

Alternatively, introduce `DoiIsbnThenSearchBasedFetcher` class, which is a wrapper around a given fetcher and tries Doi and Isbn first.

Having that class separatetly, the code can be tested independently from the gui code.

**Update** There is `org.jabref.logic.importer.CompositeIdFetcher`. Thus, just this needs to be used at Web Search.

**Pull request closed without merging**